docs/design/2023-04-11-pause-user-ddl-when-upgrading.md

```diff
@@ -0,0 +1,112 @@
+2. Notify the cluster that all TiDB enters to pause executing user DDL operations. Notification method:
+   1. New TiDB: Set the PD path (for example: `/tidb/server/global_state` , use this path later) to the current TiDB version. In addition, in order to prevent multiple TiDB nodes from upgrading at the same time, it will check whether the value on the `/tidb/server/global_state` path is empty.
+   2. All TiDBs：
+      1. The newly started TiDB will first check the `/tidb/server/global_state` value and whether it needs to enter upgrading mode.
+      2. The running TiDB will watch `/tidb/server/global_state` , and when it receives an upgrade status notification, it will enter upgrading mode.
+   3. Owner TiDB: Mark the information in the owner value on the PD (that is, the value corresponding to the DDL owner key recorded on the PD) to indicate that upgrading has been received.
+   4. New TiDB: Check whether the owner TiDB has received the upgrading signal. If so, enter the following process, otherwise wait.
+3. After receiving `/tidb/server/global_state` notification
+   1. All TiDBs: At this time, for the newly received DDL operation, consider setting it to pause directly.
+   2. Owner TiDB: For existing user DDL operations, pause them (consider removing step-3.i). Enable the pause DDL function (in implementation), and pause all user DDL operations (to distinguish between operations that the user himself has paused).
+4. New TiDB: Perform upgrade-related operations in normal bootstrap.
+5. New TiDB: Restore the original user DDL operation, that is, resume all DDL operations (the operation that the user paused before bootstrap needs to be preserved).
+6. Resume normal execution of DDL mode
+   1. New TiDB: Release pause mode on PD and set `/tidb/server/global_state` and the owner value to normal state.
+   2. All TiDBs: watch `/tidb/server/global_state` , receive this notification, and enter normal mode.
+7. Complete the bootstrap operation.
+
+![Figure 1: step 2 flow chart](./imgs/pause-user-ddl-when-upgrading-step-2.png)
+
+### Limitations
+
+In the plan, we distinguish between DDL operation types. Among them, we distinguish whether the DDL operation is a system DDL operation or a user DDL operation  by whether the DDL operation is performed on the system table. Therefore, we require users to not perform DDL operations on system tables when upgrading.
+
+## Compatibility
+
+- TiDB
+  - This feature is not supported when upgrading from v7.1 and previous versions.
+- Other components
+  - During the upgrade process, the operation of some components is not supported, such as:
+    - BR, Lightning ( Physical Import Mode ) : The operation of this type of component cannot be processed during the upgrade process, because the DDL in the paused state may be copied to TiDB during these operations. However, the DDL in this state cannot be automatically resumed, which may lead to subsequent DDL block.
```

Comment:
```suggestion
    - Backup/Restore, Lightning ( Physical Import Mode ) : The operation of this type of component cannot be processed during the upgrade process, because the DDL in the paused state may be copied to TiDB during these operations. However, the DDL in this state cannot be automatically resumed, which may lead to subsequent DDL block.
```